ec2api/api/security_group.py

```diff
@@ -56,26 +56,28 @@ def get_security_group_engine():
         return SecurityGroupEngineNova()
 
 
-def create_security_group(context, group_name, group_description,
-                          vpc_id=None):
+def create_security_group(context, group_name, group_description, vpc_id=None):
     group_name = str(group_name)
     group_description = str(group_description)
-    nova  = clients.nova(context)
-    if vpc_id and group_name != vpc_id:
-        security_groups = describe_security_groups(
-            context,
-            filter=[{'name': 'vpc-id',
-                     'value': [vpc_id]},
-                    {'name': 'group-name',
-                     'value': [group_name]}])['securityGroupInfo']
+    nova = clients.nova(context)
+# added check for mandatory parameters
+    if group_name is None or group_description is None:
+        raise exception.MissingParameter(param='group name  or group description')
```

Comment:
This check is not required here. If sg name or description is None then it will be rejected in validator routine itself.